### ISSUE TITLE
Missing key(s) in state_dict

### ISSUE THREAD

**Issue by sgsdxzy** [d8ahazard/sd_dreambooth_extension#6]
When creating a model, it reports an error:
```
RuntimeError: Error(s) in loading state_dict for CLIPTextModel:
        Missing key(s) in state_dict: "text_model.embeddings.position_ids", ...
        Unexpected key(s) in state_dict: "embeddings.position_ids", ...
```
I need to change line 713 in conversion.py from
```
text_model_dict[key[len("cond_stage_model.transformer."):]] = checkpoint[key]
```
to
```
text_model_dict["text_model." + key[len("cond_stage_model.transformer."):]] = checkpoint[key]
```
to work.

**Comment by d8ahazard** [comment]
What source checkpoint are you using?

**Comment by sgsdxzy** [comment]
> What source checkpoint are you using?

animefull-final-pruned.ckpt from NAI.

**Comment by d8ahazard** [comment]
This should now be addressed with 5a3044d, please update and let me know. ;)

**Comment by sgsdxzy** [comment]
> This should now be addressed with [5a3044d](https://github.com/d8ahazard/sd_dreambooth_extension/commit/5a3044d3f29d66c1172890ecae2a2df823537a72), please update and let me know. ;)

I think not, the error is actually returned by `text_model.load_state_dict(text_model_dict)`, so text model is expecting keys starting with `"text_model."`. The inserting to dict `text_model_dict[key[len("cond_stage_model.transformer."):]] = checkpoint[key]` would never raise any exception.

**Comment by d8ahazard** [comment]
> > This should now be addressed with [5a3044d](https://github.com/d8ahazard/sd_dreambooth_extension/commit/5a3044d3f29d66c1172890ecae2a2df823537a72), please update and let me know. ;)
> 
> I think not, the error is actually returned by `text_model.load_state_dict(text_model_dict)`, so text model is expecting keys starting with `"text_model."`. The inserting to dict `text_model_dict[key[len("cond_stage_model.transformer."):]] = checkpoint[key]` would never raise any exception.

Gotcha. I think you're right. 

Just for the sake of insanity, can you test the current commit and verify? I can rewrite again with the right try/catch if it's still failing for you.

**Comment by sgsdxzy** [comment]
> > > This should now be addressed with [5a3044d](https://github.com/d8ahazard/sd_dreambooth_extension/commit/5a3044d3f29d66c1172890ecae2a2df823537a72), please update and let me know. ;)
> > 
> > 
> > I think not, the error is actually returned by `text_model.load_state_dict(text_model_dict)`, so text model is expecting keys starting with `"text_model."`. The inserting to dict `text_model_dict[key[len("cond_stage_model.transformer."):]] = checkpoint[key]` would never raise any exception.
> 
> Gotcha. I think you're right.
> 
> Just for the sake of insanity, can you test the current commit and verify? I can rewrite again with the right try/catch if it's still failing for you.

I tested it again, and the same error still exists.

**Comment by d8ahazard** [comment]
Okay, just pushed a revised "fix", give it a go now:

[3c68fc0](https://github.com/d8ahazard/sd_dreambooth_extension/commit/3c68fc0078a02b83be93254a3774b55aed8ed2bb)

**Comment by sgsdxzy** [comment]
Fixed by [3c68fc0](https://github.com/d8ahazard/sd_dreambooth_extension/commit/3c68fc0078a02b83be93254a3774b55aed8ed2bb)
